clients/src/main/java/org/apache/kafka/common/requests/AbstractResponse.java

```diff
@@ -57,6 +59,10 @@ public ByteBuffer serialize(ApiKeys apiKey, short version, int correlationId) {
         return Collections.singletonMap(error, 1);
     }
 
+    protected Map<Errors, Integer> errorCounts(Stream<Errors> errors) {
+        return errors.collect(Collectors.groupingBy(e -> e, Collectors.summingInt(e -> 1)));
+    }
+
     protected Map<Errors, Integer> errorCounts(Collection<Errors> errors) {
```

Comment:
it seems this implementation can be replaced by ```#errorCounts(Stream<Errors>)```

Reply:
@tombentley Have we done any testing to verify that this way of doing things is no slower than the old way?

Reply:
@chia7712 you're right, but the only two remaining callers of this method are for RPCs which haven't been converted to the message generator. There's little benefit to changing them when there are already PRs for converting those RPCs, and I'm planning to remove this method entirely when those PRs have been merged. I guess I could mark this method as `@Deprecated`.

Relatedly there's only a single caller of the `apiErrorCounts(Map<?, ApiError> errors)` method which is also for an not-yet-converted RPC with a PR. If this gets merged first I'll be able to remove `apiErrorCounts(Map<?, ApiError> errors)` in that PR.